### PR TITLE
Weather makes it harder to discern sounds.

### LINKED ISSUE #1116: Rain and Bad Weather Reduce Hearing/Noise Enhancement
I think it should also reduce hearing/noise level to some extent as in the movie I watched forgot what title it was is that the assassin in the city rooftop would go killing his target with his silenced sniper rifle on heavy rain/bad weather as it would further silence the sound produced by it being muffled out in the rain/thunder storm 
(Also read in a book forgot title though that a suppressed .22 caliber gun in heavy rain is completely/very silent even at point blank range)

This would change game-play strategy as firing suppressed guns in heavy rain/bad weather would allow a more stealthy play style.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1366927-rain-and-bad-weather-reduce-hearing-noise-enhancement?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by i2amroy**
Weather now makes it more difficult for the player (and zombies) to discern sounds. It does so by reducing the volume by a weather factor after any physical effects (since it raining won't stop you from going deaf) but prior to any attention-based checks. This means that it'll be easier to sneak around in the clatter of a thunderstorm than on a calm day. Ideally we'll eventually have modifiers based on inside/outside as well, but since the sound code is rather heavily used I didn't want to attempt anything too expensive performance wise.

Note: Snow gets a sound reduction here not necessarily because it's loud as it falls, but because it is very good at absorbing and muting sound.

Closes #1116

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
Awesome!

**Comment by Lain-** [comment]
Agree. Sneaking around in the rain now becomes a reality.

File: src/sounds.cpp [hunk 1]
```diff
@@ -204,12 +209,21 @@ void sounds::process_sound_markers( player *p )
             continue;
         }
 
+        // At this point we are dealing with attention (as opposed to physical effects)
+        // so reduce volume by the amount of ambient noise from the weather.
+        const int mod_vol = volume - weather_vol;
+
+        // The noise was drowned out by the surroundings
+        if (mod_vol <= 0) {
```
Comment (BevapDin) [inline src/sounds.cpp]: Can you please change this to `< 0`, there are some sounds in iuse.cpp (bombs ticking) that use volume=0 and they should be heard in silent weather. Or add another check to skip the sound if it's 0 *and* weather_vol is not 0. With the current comparison, the bomb ticking is never heard.

**Comment by Draconomial** [comment]
👍

File: src/sounds.cpp (hunk 1 above)
Comment (i2amroy) [inline src/sounds.cpp]: Ah I had missed that. I'll probably do the latter, since you shouldn't be hearing sounds that were dropped to 0.

**Pull request merged**